app-creator/client/src/utils/helpers.ts

```diff
@@ -103,3 +113,21 @@ export function deepCloneTemplate(
 
   return clone;
 }
+
+/**
+ * Takes a snapshot of the current template and stores it in localStorage.
+ */
+export function storeTemplateInLocalStorage() {
+  try {
+    /**
+     * Saving current template string in localStorage so we can transfer data across.
+     * */
+    const currentTemplate = JSON.stringify(
+      deepCloneTemplate(store.getState().template)
+    );
+
+    localStorage.setItem('previousTemplate', currentTemplate);
```

Comment:
Here are two ways we can solve this: 
1. We store an object with the timestamp as the key and the template string as the value. We open a new app instance passing in the timestamp as a URL param. We then retrieve the corresponding template given the timestamp. 
2. We can read the value from local storage as soon as the app renders. Since the user can't duplicate two apps simultaneously, there won't be a race condition. The current implementation reads the template on the `continue` action on the template wizard, as a result, duplicating a second app on another window before clicking `continue` on the previous one will lead to the template being overwritten.